### DISCUSSION TITLE
The api call  before msw has been activated

### DISCUSSION THREAD

**Discussion by anhuet** [mswjs/msw#1270] [Q&A]
I am having trouble getting msw to intercept my requests.
When I refresh react(nextjs) page, the api query in useEffect was called, so it returned the null data because msw has not been activated.
Is there a way to active msw before this api call?

**Comment by kettanaito** [comment]
Hey, @anhuet. 

Please include a reproduction repository for this. It's hard for me to advise anything without seeing how you're using MSW, which versions you're using etc. MSW defers all requests that happen during the worker registration and resolves them after. If this doesn't happen in your case, I'd like to look at that case. 

You can also use [Deferred mounting](https://mswjs.io/docs/recipes/deferred-mounting) of your application but I recommend it only as a last resort.
